docs/tutorial/native-code-and-electron.md

```diff
@@ -0,0 +1,375 @@
+# Native Code and Electron
+
+One of Electron's most powerful features is the ability to combine web technologies with native code - both for compute-intensive logic as well as for the occasional native user interface, where desired.
+
+Electron does so by building on top of "Native Node.js Addons". You've probably already come across a few of them - packages like the famous [sqlite](https://www.npmjs.com/package/sqlite3) use native code to combine JavaScript and native technologies. You can use this feature to extend your Electron application with anything a fully native application can do:
+
+* Access native platform APIs not available in JavaScript. Any macOS, Windows, or Linux operating system API is available to you.
+* Create UI components that interact with native desktop frameworks.
+* Integrate with existing native libraries.
+* Implement performance-critical code that runs faster than JavaScript.
+
+Native Node.js addons are dynamically-linked shared objects (on Unix-like systems) or DLL files (on Windows) that can be loaded into Node.js or Electron using the `require()` or `import` functions. They behave just like regular JavaScript modules but provide an interface to code written in C++, Rust, or other languages that can compile to native code.
+
+# Tutorial: Creating a Native Node.js Addon for Electron
```

Comment:
Should we specify that this is C++ specifically in the heading here?

Reply:
I'm hesitant to do that because the steps in this tutorial will be the beginning of all other tutorials. The last step, the implementation file, will be slightly different with other technologies, but I will otherwise not repeat the core steps in every tutorial.

In other words, I'd like to keep a language out of the title clear from assumptions about which language you want to use later. 

Maybe I'll add language to the very last step.